src/Serializer/ItemNormalizer.php

```diff
@@ -13,54 +13,11 @@
 
 namespace ApiPlatform\Core\Serializer;
 
-use ApiPlatform\Core\Exception\InvalidArgumentException;
-
 /**
  * Generic item normalizer.
  *
  * @final
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
  */
-class ItemNormalizer extends AbstractItemNormalizer
-{
-    /**
-     * {@inheritdoc}
-     *
-     * @throws InvalidArgumentException
-     */
-    public function denormalize($data, $class, $format = null, array $context = [])
-    {
-        // Avoid issues with proxies if we populated the object
-        if (isset($data['id']) && !isset($context[self::OBJECT_TO_POPULATE])) {
-            if (isset($context['api_allow_update']) && true !== $context['api_allow_update']) {
-                throw new InvalidArgumentException('Update is not allowed for this operation.');
-            }
-
-            $this->updateObjectToPopulate($data, $context);
-        }
-
-        return parent::denormalize($data, $class, $format, $context);
-    }
-
-    private function updateObjectToPopulate(array $data, array &$context)
-    {
-        try {
-            $context[self::OBJECT_TO_POPULATE] = $this->iriConverter->getItemFromIri((string) $data['id'], $context + ['fetch_data' => true]);
```

Comment:
This whole method was nonsense imho. Looks like it was developed before there was the `IdentifiersExtractor`.

Reply:
> Looks like it was developed before there was the `IdentifiersExtractor`.

It's the case.

Reply:
You should add a deprecated error here saying this class is deprecated, and "unabstract" the `AbstractItemNormalizer`

Reply:
I don't think that's a good plan. I'd keep it as is because the `AbstractItemNormalizer` is not used directly anywhere now and maybe there will be the day we have to add something to the `ItemNormalizer`. It can still be refactored for 3.0 even if there's no deprecation message.